core/src/commonMain/kotlin/com/xebia/functional/xef/auto/AIRuntime.kt

```diff
@@ -0,0 +1,32 @@
+package com.xebia.functional.xef.auto
+
+import com.xebia.functional.xef.embeddings.OpenAIEmbeddings
+import com.xebia.functional.xef.env.OpenAIConfig
+import com.xebia.functional.xef.llm.openai.KtorOpenAIClient
+import com.xebia.functional.xef.llm.openai.LLMModel
+import com.xebia.functional.xef.vectorstores.LocalVectorStore
+import kotlin.jvm.JvmStatic
+import kotlin.time.ExperimentalTime
+
+data class AIRuntime<A>(val runtime: suspend (block: AI<A>) -> A) {
+  companion object {
+    @OptIn(ExperimentalTime::class)
+    @JvmStatic
+    fun <A> openAI(): AIRuntime<A> = AIRuntime { block ->
```

Comment:
I guess this is something that we will move to a specific module for OpenAI so we don't include a dependency to OpenAI in the core library, won't we?

Reply:
Yes, this PR does not address moving open ai deps out of the core. We still need to move the open ai dependencies out of the core. It's just the first step toward that, trying to cause as few file movements as possible. We plan on using the open ai models as a baseline interface for the rest of the models to adapt to them.